--- a/scripts/admin-update.sh	2026-04-23 11:27:16.760404851 -0400
+++ b/scripts/admin-update.sh	2026-04-24 00:11:22.113533356 -0400
@@ -585,18 +585,13 @@
 
 DIR="$PWD"
 cd $GIT_TREE
-GIT_DESCRIBE=$(git describe)
-GIT_BRANCH=$(git branch --no-color |sed -n 's/^\* //p')
-GIT_BRANCH=${GIT_BRANCH:-master}
-GIT_REMOTE=$(git config branch.${GIT_BRANCH}.remote)
-GIT_REMOTE=${GIT_REMOTE:-origin}
-GIT_REMOTE_URL=$(git config remote.${GIT_REMOTE}.url)
-GIT_REMOTE_URL=${GIT_REMOTE_URL:-unknown}
+GIT_DESCRIBE=ol_R5.SP4.01
+GIT_REMOTE_URL=https://github.com/TI-OpenLink/wl12xx.git
 echo -e "${GREEN}Updated${NORMAL} from local tree: ${BLUE}${GIT_TREE}${NORMAL}"
 echo -e "Origin remote URL: ${CYAN}${GIT_REMOTE_URL}${NORMAL}"
 
 echo -e "Getting TI drivers version..."
-TI_VERSION=`git describe --dirty`
+TI_VERSION=ol_R5.SP4.01
 TI_DRIVERS="wl12xx"
 TI_PATH=drivers/net/wireless
 echo -e "TI drivers version: ${TI_VERSION}"
@@ -607,11 +602,11 @@
 	${DIR}/${TI_PATH}/${driver}/version.h
 done
 cd $DIR
-if [ -d ./.git ]; then
+# if [ -d ./.git ]; then
 	if [[ ${POSTFIX_RELEASE_TAG} != "" ]]; then
 		echo -e "$(git describe)-${POSTFIX_RELEASE_TAG}" > compat_version
 	else
-		echo -e "$(git describe)" > compat_version
+		echo -e "ol_R5.SP4.01" > compat_version
 	fi
 
 	cd $GIT_TREE
@@ -651,6 +646,6 @@
 	echo -e "compat-wireless release: ${YELLOW}$(cat compat_version)${NORMAL}" >> $CODE_METRICS
 
 	cat $CODE_METRICS
-fi
+# fi
 
 ./scripts/driver-select restore
